packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -159,14 +159,16 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
 
         uint256 _attoSharesOwned = shareToken.lowestBalanceOfMarketOutcomes(_market, _shortOutcomes, _owner);
 
-        uint256 _attoSharesPurchasable = cash.balanceOf(_owner).div(_price);
+        uint256 _availableCash = cash.allowance(_owner, augur).min(cash.balanceOf(_owner));
+        uint256 _attoSharesPurchasable = _availableCash.div(_price);
 
         return _attoSharesOwned.add(_attoSharesPurchasable);
     }
 
     function askBalance(address _owner, IMarket _market, uint8 _outcome, uint256 _price) public view returns (uint256) {
         uint256 _attoSharesOwned = shareToken.balanceOfMarketOutcome(_market, _outcome, _owner);
-        uint256 _attoSharesPurchasable = cash.balanceOf(_owner).div(_market.getNumTicks().sub(_price));
+        uint256 _availableCash = cash.allowance(_owner, augur).min(cash.balanceOf(_owner));
```

Comment:
This should be the allowance of FillOrder and not augur

Reply:
Yup! Thanks for the catch

Reply:
I wish there was a standard for "how much can i transferFrom" without calling into the contract twice.